hrms/hr/doctype/shift_assignment/shift_assignment.py

```diff
@@ -377,7 +377,7 @@ def get_employee_shift_timings(
 	curr_shift = get_employee_shift(employee, for_timestamp, consider_default_shift, "forward")
 	if curr_shift:
 		next_shift = get_employee_shift(
-			employee, curr_shift.start_datetime + timedelta(days=1), consider_default_shift, "forward"
+			employee, for_timestamp + timedelta(days=1), consider_default_shift, "forward"
```

Comment:
This original logic is correct because the next shift is after the current shift. So it's relative to the current shift's timestamp?

What's the issue you are exactly facing? Can you update screenshots for the complete scenario and what's not working as expected?

Reply:
I had a use case where a company has 2 12 hours shift

1. Day shift ( Timing 7:00 am to 7:00 pm )
2. Night shift ( Timing 7:00 pm to 7:00 am )

Both have Allowed Checkin after Shift time of 240 minutes

The issue comes when Current Shift and Next Shift are not the same.
E.g
An employee has a Day shift on April 1st and a Night shift on April 2nd
Employee check-in type In 2023-04-01 07:00:00 and type out 2023-04-01 07:03:56
1st check-in was marked against Day shit but 2nd check-in was against the night shift.

When I dig deep and the function which was getting the next shift was returning the wrong next shift times you can try this use case

Reply:
https://github.com/frappe/hrms/pull/532

This PR fixes it. Closing